department-of-veterans-affairs/abd-vro
# Changes BIP API to implement passthrough behavior.

Implements the getClaimContentions endpoint as more of a passthrough implementation of the response received from the BIP API. This class refactors the return object used by the BipApiService and the return object of the corresponding RabbitListener.

## What was the problem?

Current implementation does not explicitely return the exact content and responses received from the BIP API. It obfuscates all non 2xx responses and returns a 500 to the client over rabbitmq. 

Associated tickets or Slack threads:
- #2091 
- [slack thread](https://dsva.slack.com/archives/C01Q7979Z7D/p1698274813957249)
- Blocked by:
  - #2144 

## How does this fix it?[^1]

This PR creates a new pattern for the structure of returned objects. This allows all responses (good (2xx) or bad(not 2xx)) from BIP API to be returned to client of `svc-bip-api`. This structure also allows returning 500s as a response when `svc-bip-api` is the culprit of the error.
<img width="921" alt="image" src="https://github.com/department-of-veterans-affairs/abd-vro/assets/135860892/8efe4313-21a2-48e7-8da9-8d7ea1342b93">


## How to test this PR
- Run the unit tests via `./gradlew :svc-bip-api:check`
- Start the platform base docker containers to start RabbitMq and start the BIP Mock container (see [Docker Compose](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose))
  - run the integration tests via `./gradlew :svc-bip-api:integrationTest`
  - build and compose the `svc-bip-api` docker and run a test by:
     - create a new queue on the `bip-api` exchange called `response`
     - open to the rabbitmq console to the [getClaimContentionsQueue](http://localhost:15672/#/queues/%2F/getClaimContentionsQueue) 
     - publish a message with the body `{"claimId":1010}` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 and the response is the mocked response


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
